tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/thread/thread.component.ts

```diff
@@ -0,0 +1,168 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+import { Comment, Thread } from '@/shared/proto';
+import { AuthService } from '@/shared/services';
+import { ThreadState, ThreadStateService } from './thread-state.service';
+
+// The component implements a single thread.
+// How it looks: https://i.imgur.com/WGPp361.jpg
+@Component({
+  selector: 'cr-thread',
+  templateUrl: './thread.component.html',
+  styleUrls: ['./thread.component.scss'],
+})
+export class ThreadComponent implements OnInit, OnDestroy {
+  isCommentOpenMap: boolean[] = [];
+  isReply: boolean = false;
+  textarea: FormControl = new FormControl();
+  resolvedCheckbox: FormControl = new FormControl();
+  subscription = new Subscription();
```

Comment:
Let's put type enforcement here

Reply:
I think type enforcement is redundant with `new`.
We already tell anybody who review the code that type is `Subscription`. Code editor or IDE also know that it's `Subscription` even without new.
So let's delete type enforcement to `textarea` and `resolvedCheckbox`

Reply:
@MateMahinKhan Also, do you know a way to use typedef option of tslint, but to not force type with new?